### ISSUE TITLE
gnt-network connect with invalid mode raises invalid exception

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#377]
Originally reported of Google Code with ID 321.

```
[cluster] root@node4:~# gnt-network connect test hot a a
Corrupt configuration file: Invalid nic mode: a
Aborting.

That's because the mode is not verified at opcode level, being declared simply as string. We might want to add more checks in general to the opcodes and/or better exception raising.
```
Originally added on 2012-11-21 17:40:26 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2012-12-18 16:20:59 +0000 UTC.

Added to Milestone: Release2.7

**Comment by Ganeti-Issues-Migrator** [comment]
```
Commit 6ef3739 added verification of the network mode at opcode level and commit e896ad3 added bash completion for the mode.

# gnt-network connect test hot a a
…
Parameter 'OP_NETWORK_CONNECT.network_mode' fails validation

# gnt-network connect test hot <tab>
bridged      openvswitch  routed
```
Originally added on 2012-12-19 18:05:08 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Marking 2.7 bugs fixed, as 2.7.0~rc1 was released, and should be ready to be tested in production environments.
```
Originally added on 2013-05-09 05:42:48 +0000 UTC.

Changed State: Released
